### PR TITLE
Undo five commits

### PR DESCRIPTION
**Pull request by richsalz**

### CODE REVIEW

**Comment by richsalz** [comment]
For details, please see the commentary on each commit.
I do *not* planning on squashing this if it gets merged.

**Comment by richsalz** [comment]
see also https://www.openssl.org/blog/blog/2017/06/17/code-removal/

File: crypto/err/err.c [hunk 1]
```diff
@@ -749,6 +749,8 @@ void ERR_add_error_vdata(int num, va_list args)
     n = 0;
     for (i = 0; i < num; i++) {
         a = va_arg(args, char *);
+        if (a == NULL)
+            a = "<NULL>";
```
Comment (levitte) [inline crypto/err/err.c]: The other option, which I was considering, is this:

``` C
        if (a == NULL)
            continue;
```

**Comment by richsalz** [comment]
I think pushing NULL on the error data is really a programming error, and I wanted to be able to find them not ignore them.
This fix was easy, my first attempt to run the tests had a crash in strlen where “a” was NULL.

**Comment by levitte** [comment]
Hmmm, and you didn't try to fix the tests that are giving it NULL parameters?

On the other hand, you're right, this is pretty much the same as `printf` does when given NULL parameters.

**Comment by richsalz** [comment]
Fixing them would be a better fix.  But this PR is just about removing certain commits.

**Comment by kaduk** [comment]
A nit, but did you consider "revert" vs. "undo" in the commit messages?

**Comment by richsalz** [comment]
Yes we did.  Revert has a specific/common meaning in git, and this was not that.

**Comment by richsalz** [comment]
FYI, config with enable-asan and enable-msan, and running all tests (including fuzz) came out clean.

**Comment by richsalz** [comment]
@dot-asm, can you explain the travis failures?  Not what I expected.

**Comment by levitte** [comment]
@mattcaswell mentioned this a couple of days ago.  The working assumption so far is that there's a bug in clang 3.9, or perhaps locally at Travis'.

**Comment by richsalz** [comment]
thank you, done.

**Pull request closed without merging**